### PR TITLE
add shortname for istio crds

### PR DESCRIPTION
**Pull request by GregHanson**
followup to this closed PR: https://github.com/istio/istio/pull/3394

Resolves https://github.com/istio/istio/issues/2650

Also not sure why gateways were neglected in one of the template files

### CODE REVIEW

**Comment by GregHanson** [comment]
cc @christian-posta

File: install/kubernetes/helm/istio/charts/pilot/templates/crds.yaml [hunk 1]
```diff
@@ -75,6 +76,7 @@ spec:
     listKind: DestinationRuleList
     plural: destinationrules
     singular: destinationrule
+    shortNames: ["drule"]
```
Comment (rshriram) [inline install/kubernetes/helm/istio/charts/pilot/templates/crds.yaml]: destrule

File: install/kubernetes/helm/istio/charts/pilot/templates/crds.yaml [hunk 2]
```diff
@@ -91,6 +93,7 @@ spec:
     listKind: ServiceEntryList
     plural: serviceentries
     singular: serviceentry
+    shortNames: ["sent"]
```
Comment (rshriram) [inline install/kubernetes/helm/istio/charts/pilot/templates/crds.yaml]: dude!! :).. 
svcent

**Comment by rshriram** [comment]
please change sent -> svcent, and drule to destrule or dstrule..

**Comment by frankbu** [comment]
+1 on `svcent`
I prefer `dstrule` over `destrule`

**Comment by ayj** [comment]
Would shell completion address the same problem (see https://github.com/istio/istio/issues/6119)? It would avoid name ambiguity issues raised in the original issue. My understanding is that short names are primarily used for interactive sessions, e.g. demos.

**Comment by GregHanson** [comment]
@ayj I like the idea of bash completion, but at the same time short names are something kubernetes has already implemented.  Is there a reason we shouldn't follow the standard kubernetes already has in place?

**Comment by ayj** [comment]
Just because CRD allow short names doesn't mean we should use them :) I'd like to understand if we really need them. They are certainly convenient but introduce the possibility of ambiguity / conflict. Non-fully qualified names and short names in particular are context sensitive but CRD naming is cluster scoped. We've already had one naming conflict (see https://github.com/istio/istio/issues/5532).

**Comment by ijsnellf** [comment]
My two cents: most of these shortnames are kind of long. I like the one for gateway, `gw`, but `serviceentry`->`svcent` doesn't remove a whole lot of characters. Are we avoiding super short shortnames like `serviceentry` -> `se` or `virtualservice` -> `vs`?

**Comment by rshriram** [comment]
if we do go the short name route, I like @ijsnellf 's suggestion of vs, dr, gw, se,

**Comment by frankbu** [comment]
I'm worried about both conflicts and being overly cryptic given the global context of these names. What, for example, might `dr` be in a global context? I think the names that @GregHanson suggested are consistent with the approach in k8s, i.e., names that are pretty intuitive like `svc` is short for `service`, `vsvc` is short for `virtualservice`.

Anyway, that said, maybe @ayj is right, and we shouldn't do this if bash completion solves the problem? On the other hand, we could try experimenting with them now to see how it goes and then we can still decide to remove them before 1.0.

File: pilot/pkg/model/config.go [hunk 3]
```diff
@@ -229,6 +232,16 @@ func (descriptor ConfigDescriptor) GetByMessageName(name string) (ProtoSchema, b
 	return ProtoSchema{}, false
 }
 
+// GetByShortName finds a schema by short name if it is available
+func (descriptor ConfigDescriptor) GetByShortName(name string) (ProtoSchema, bool) {
```
Comment (gyliu513) [inline pilot/pkg/model/config.go]: I found this was only in test, can this be moved to test file?

**Pull request closed without merging**